Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/SharedOpaqueDirectoryTests.cs

```diff
@@ -1257,6 +1257,36 @@ public void AbsentPathProbeInUndeclaredOpaquesUnsafeModeCachedPip(bool forceDepe
             }
         }
 
+        [Fact]
+        public void ReadWrittenFileUnderSharedOpaqueWithWhitelistIsAllowed()
+        {
+            string sharedOpaqueDir = Path.Combine(ObjectRoot, "sod");
+            AbsolutePath sharedOpaqueDirPath = AbsolutePath.Create(Context.PathTable, sharedOpaqueDir);
+            DirectoryArtifact sharedOpaqueDirArtifact = DirectoryArtifact.CreateWithZeroPartialSealId(sharedOpaqueDirPath);
+            FileArtifact outputInSharedOpaqueDir = CreateOutputFileArtifact(root: sharedOpaqueDir, prefix: "sod-file");
+            FileArtifact sourceFile = CreateSourceFile();
+
+            FileArtifact whitelistFile = CreateSourceFile();
+            global::BuildXL.Utilities.Configuration.Mutable.FileAccessWhitelistEntry entry = new global::BuildXL.Utilities.Configuration.Mutable.FileAccessWhitelistEntry()
+            {
+                Value = "testValue",
+                PathFragment = ArtifactToString(whitelistFile),

```

Comment:
>whitelistFile [](start = 48, length = 13)

It doesn't look like the test pip is interacting with this file. Maybe I'm missing the point of the test?

Reply:
Good catch. At first I thought that the accesses were not reported because I didn't have whitelist entries. I'm removing it.

---
In reply to: [314461083](https://github.com/microsoft/BuildXL/pull/741#discussion_r314461083) [](ancestors = 314461083)